ci: update the mergify rule to exclude k8s 1.25

yati1998 opened the pull request:
This commit excludes the k8s 1.25 check from the
mergify rules for ready-to-merge label

yati1998 committed:
ci: update the merfigy rule to exclude k8s 1.25

This commit excludes the k8s 1.25 check from the
mergify rules for ready-to-merge label

Signed-off-by: Yati Padia <ypadia@redhat.com>

nixpanic reviewed (commented):
This looks OK, but it assumes there will not be a Ceph-CSI 3.9.x update anymore. Are you certain that 3.9 is EOL (and that 3.10 is the oldest maintained version)?

yati1998 committed:
ci: remove the merfigy rule for devel branch

this commit removes the automatic merge rule
for devel branch as this is always overwritten
from the conditions above and is not required

Signed-off-by: Yati Padia <ypadia@redhat.com>

nixpanic commented on .mergify.yml:
```diff
@@ -355,31 +355,6 @@ pull_request_rules:
               - "status-success=ci/centos/upgrade-tests-cephfs"
               - "status-success=ci/centos/upgrade-tests-rbd"
               - "status-success=DCO"
```
Ineeed, the **devel + `ready-to-merge`** rule will never be met, as GitHub requires 2 (real) reviews for PRs on the devel branch, Mergify is aware of that requirement automatically.

nixpanic commented:
@mergifyio refresh
